samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemProducer.scala

```diff
@@ -125,31 +97,17 @@ class KafkaSystemProducer(systemName: String,
       throw new IllegalArgumentException(StreamNameNullOrEmptyErrorMsg)
     }
 
-    val sourceData = sources.get(source)
-    if (sourceData == null) {
+    if (!sources.contains(source)) {
```

Comment:
Just wondering: This happens on every send. Does this still provide value besides a defensive check against our (i.e. framework devs as opposed to user) coding errors?

Reply:
Another great question. I never bothered to look into the history to see whether this check was to enforce the proper state transitions or if it was just to make sure the map is prepopulated in a synchronous way. Looks like it provides no value now. I'll remove it!